extensions/HyperVExtension/src/HyperVExtension/Models/ByteTransferProgress.cs

```diff
@@ -14,9 +14,23 @@ public class ByteTransferProgress
 
     public uint PercentageComplete => (uint)((BytesReceived / (double)TotalBytesToReceive) * 100);
 
+    public string ErrorMessage { get; set; } = string.Empty;
+
     public ByteTransferProgress(long bytesReceived, long totalBytesToReceive)
     {
         BytesReceived = bytesReceived;
         TotalBytesToReceive = totalBytesToReceive;
     }
+
+    public ByteTransferProgress(string errorMessage)
+    {
+        ErrorMessage = errorMessage;
+    }
+
+    public bool CompletedSuccessfully => BytesReceived == TotalBytesToReceive;
+
+    public bool Failed()
+    {
+        return !string.IsNullOrEmpty(ErrorMessage);
+    }
```

Comment:
If you create an instance of this class using ByteTransferProgress(string errorMessage) then both CompletedSuccessfully and Failed will return true which seems incorrect and makes usage error prone.
Consider a single Status property that would return an enum value, so it's always unambiguous and can be extended (Completed, Failed, InProgress, Paused, etc.).

Reply:
Sure I can do that